File: tests/graph/eval/fixtures/spring_boot_sample/src/main/java/com/example/blog/controller/UserController.java
package com.example.blog.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.blog.dto.UserDto;
import com.example.blog.entity.User;
import com.example.blog.service.UserService;

@RestController
@RequestMapping("/api/users")
public class UserController {
    private final UserService userService;

    @Autowired
    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping
    public UserDto register(@RequestBody UserDto payload) {
        User user = userService.register(payload.email, payload.name);
        return UserDto.from(user);
    }

    @GetMapping("/{id}")
    public UserDto get(@PathVariable Long id) {
        return UserDto.from(userService.get(id));
    }
}

File: tests/graph/eval/fixtures/kotlin_spring_sample/src/main/kotlin/com/example/blog/controller/UserController.kt
package com.example.blog.controller

import com.example.blog.entity.User
import com.example.blog.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

data class RegisterPayload(val email: String, val name: String)

@RestController
@RequestMapping("/api/users")
class UserController(private val userService: UserService) {

    @PostMapping
    fun register(@RequestBody payload: RegisterPayload): User =
        userService.register(payload.email, payload.name)

    @GetMapping("/{id}")
    fun read(@PathVariable id: Long): User? = userService.get(id)
}
